code/forms/JsonField.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * Class JsonField combines form inputs into a key-value pair
+ */
+class JsonField extends CompositeField
+{
+    public function __construct($name, $children = null)
+    {
+        $this->setName($name);
+
+        if ($children instanceof FieldList) {
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        } elseif (is_array($children)) {
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        } else {
+            $children = is_array(func_get_args()) ? func_get_args() : array();
+            if (!empty($children)) {
+                array_shift($children);
+            }
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        }
+        parent::__construct($children);
+    }
+
+    public function hasData()
+    {
+        return true;
+    }
+
+    /**
+     * Have to has it to overwritten CompositeField behavior
+     *
+     * @param      $list
+     * @param bool $saveableOnly
+     */
+
+    public function collateDataFields(&$list, $saveableOnly = false)
+    {
+    }
+
+    function array_filter_empty_recursive($haystack)
+    {
+        foreach ($haystack as $key => $value) {
+            if (is_array($value)) {
+                $haystack[$key] = $this->array_filter_empty_recursive($haystack[$key]);
+            }
+            if (empty($haystack[$key])) {
+                unset($haystack[$key]);
+            }
+        }
+
+        return $haystack;
+    }
+
+    public function dataValue()
+    {
+        if (is_array($this->value)) {
+            $this->value = $this->array_filter_empty_recursive($this->value);
+            if (!empty($this->value)) {
+                return Convert::array2json($this->value);
+            } else {
+                return null;
+            }
+        } else {
```

Comment:
* Can you add a docblock explaining why you've overridden the parent
* Don't need an `else` here